--- orig/Sch49.ldf	2017-10-03 15:53:31.162142476 +1300
+++ Sch49.ldf	2017-10-03 16:17:07.326514822 +1300
@@ -669,26 +669,20 @@
 replace: defaultHidingValue
 defaultHidingValue: TRUE
 -  
 
 dn: CN=ms-DS-Resource-Properties,CN=Schema,CN=Configuration,DC=X
 changetype: ntdsSchemaModify
 replace: defaultHidingValue
 defaultHidingValue: TRUE
 -  
 
-dn: CN=ms-DS-List-Of-Claim-Types,CN=Schema,CN=Configuration,DC=X
-changetype: ntdsSchemaModify
-replace: defaultHidingValue
-defaultHidingValue: TRUE
--  
-
 dn: CN=ms-DS-Claim-Type,CN=Schema,CN=Configuration,DC=X
 changetype: ntdsSchemaModify
 add: systemMayContain
 systemMayContain: 1.2.840.113556.1.4.2157
 systemMayContain: 1.2.840.113556.1.4.2158
 systemMayContain: 1.2.840.113556.1.4.2098
 systemMayContain: 1.2.840.113556.1.4.2159
 systemMayContain: 1.2.840.113556.1.4.2160
 -  
 

