--- orig/Sch59.ldf	2023-02-02 11:19:49.258058871 +0100
+++ patched/Sch59.ldf	2023-02-03 12:01:09.935996927 +0100
@@ -160,23 +160,20 @@
 changetype: ntdsSchemaModify
 replace: isSingleValued
 isSingleValued: TRUE
 -
 
 dn: CN=ms-DS-Device-Physical-IDs,CN=Schema,CN=Configuration,DC=X
 changetype: ntdsSchemaModify
 replace: omSyntax
 omSyntax: 64
 -
-
-dn: CN=ms-DS-Device-Physical-IDs,CN=Schema,CN=Configuration,DC=X
-changetype: ntdsSchemaModify
 replace: attributeSyntax
 attributeSyntax: 2.5.5.12
 -
 
 dn: CN=ms-DS-Device-Physical-IDs,CN=Schema,CN=Configuration,DC=X
 changetype: ntdsSchemaModify
 replace: rangeUpper
 rangeUpper: 1024
 -
 
