--- samba-2.2.4/source/configure.in.orig	Thu May  2 00:10:02 2002
+++ samba-2.2.4/source/configure.in	Thu May  2 00:10:17 2002
@@ -380,6 +380,28 @@
 #endif
 }
 ], [LINUX_LFS_SUPPORT=yes], [LINUX_LFS_SUPPORT=no], [LINUX_LFS_SUPPORT=cross])
+	if test x$LINUX_LFS_SUPPORT = xyes ; then
+          AC_TRY_RUN([
+#include <unistd.h>
+#include <sys/types.h>
+#include <fcntl.h>
+main() {
+   unsigned int *padding;
+   struct flock foo_lock = {F_WRLCK, SEEK_SET, 0, 1, 0};
+   int fd = open("/dev/null", O_RDWR);
+
+   /* Yes, we're depending on the internals of the Linux flock structure
+      here -- but this test is explicitly Linux-specific to begin with. */
+   padding = (unsigned int *)&foo_lock;
+   padding[1] = 0xffffffff;
+   foo_lock.l_start = 0;
+   if (fcntl(fd, F_SETLK, &foo_lock) < 0)
+      exit(1);
+
+   exit(0);
+}
+], [LINUX_LFS_SUPPORT=yes], [LINUX_LFS_SUPPORT=no], [LINUX_LFS_SUPPORT=cross])
+	fi
         CPPFLAGS="$old_CPPFLAGS"
         if test x$LINUX_LFS_SUPPORT = xyes ; then
           CPPFLAGS="-D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -D_GNU_SOURCE $CPPFLAGS"
