--- samba/source/Makefile.in.orig	Thu May  2 14:03:45 2002
+++ samba/source/Makefile.in	Thu May  2 14:03:45 2002
@@ -709,9 +709,10 @@
 	@$(SHELL) $(srcdir)/script/installswat.sh $(SWATDIR) $(srcdir)
 
 installclientlib:
-	-$(INSTALLCMD) bin/libsmbclient.so
-	-$(INSTALLCMD) -d ${prefix}/include
-	-$(INSTALLCMD) include/libsmbclient.h ${prefix}/include
+	-$(INSTALLCMD) bin/libsmbclient.so $(BASEDIR)/lib
+	-$(INSTALLCMD) bin/libsmbclient.a $(BASEDIR)/lib
+	-$(INSTALLCMD) -d $(INCLUDEDIR)
+	-$(INSTALLCMD) include/libsmbclient.h $(INCLUDEDIR)
 
 # revert to the previously installed version
 revert:
