diff -ur samba-3.0.0beta1.orig/source/python/setup.py samba-3.0.0beta1/source/python/setup.py
--- samba-3.0.0beta1.orig/source/python/setup.py	2003-06-07 12:57:36.000000000 -0500
+++ samba-3.0.0beta1/source/python/setup.py	2003-06-19 17:26:48.000000000 -0500
@@ -57,6 +57,9 @@
     if lib[0:2] == "-L":
         library_dirs.append(lib[2:])
         continue
+    if lib[0:4] == "-Wl,":
+        print "Ignoring linker flag '%s' in $LIBS variable passed to setup.py" % lib
+        continue
     print "Unknown entry '%s' in $LIBS variable passed to setup.py" % lib
     sys.exit(1)
 
