diff -uNr samba-3.0alpha21.orig/source/smbadduser samba-3.0alpha21/source/smbadduser
--- samba-3.0alpha21.orig/source/smbadduser	2003-02-16 13:01:45.000000000 -0600
+++ samba-3.0alpha21/source/smbadduser	2003-02-16 16:25:16.000000000 -0600
@@ -2,13 +2,14 @@
 #
 # smbadduser - Written by Mike Zakharoff
 #
+# Customized for Debian by Eloy A. Paris <peloy@debian.org>
+#
 unalias *
-set path = ($path /usr/local/samba/bin)
+# No need to set a path in Debian
+#set path = ($path /usr/local/samba/bin)
 
-set smbpasswd = /usr/local/samba/private/smbpasswd
-#set smbpasswd = /etc/samba/smbpasswd
-set user_map  = /usr/local/samba/lib/users.map
-#set user_map  = /etc/samba/smbusers
+set smbpasswd = /etc/samba/smbpasswd
+set user_map  = /etc/samba/users.map
 #
 # Set to site specific passwd command
 #
