--- samba-2.2.4/source/client/client.c.orig	Wed May  1 23:13:57 2002
+++ samba-2.2.4/source/client/client.c	Thu May  2 00:34:16 2002
@@ -91,10 +91,10 @@
 extern file_info def_finfo;
 
 /* timing globals */
-int get_total_size = 0;
-int get_total_time_ms = 0;
-int put_total_size = 0;
-int put_total_time_ms = 0;
+SMB_BIG_UINT get_total_size = 0;
+unsigned int get_total_time_ms = 0;
+SMB_BIG_UINT put_total_size = 0;
+unsigned int put_total_time_ms = 0;
 
 /* totals globals */
 static double dir_total;
