--- samba-2.2.2.cvs20020120/source/client/client.c.orig	Wed Jan 23 23:32:44 2002
+++ samba-2.2.2.cvs20020120/source/client/client.c	Wed Jan 23 23:33:50 2002
@@ -92,10 +92,10 @@
 extern file_info def_finfo;
 
 /* timing globals */
-int get_total_size = 0;
-int get_total_time_ms = 0;
-int put_total_size = 0;
-int put_total_time_ms = 0;
+SMB_BIG_UINT get_total_size = 0;
+unsigned int get_total_time_ms = 0;
+SMB_BIG_UINT put_total_size = 0;
+unsigned int put_total_time_ms = 0;
 
 /* totals globals */
 static double dir_total;
