diff -uNr samba-orig/source/Makefile.in samba/source/Makefile.in
--- samba-orig/source/Makefile.in	Tue Jan 19 19:51:09 2000
+++ samba/source/Makefile.in	Wed Jan 20 02:04:50 2000
@@ -27,7 +27,7 @@
 # the previous releases of Samba
 SBINDIR = @bindir@
 LIBDIR = @libdir@
-VARDIR = $(BASEDIR)/var
+VARDIR = /var
 MANDIR = @mandir@
 
 # The permissions to give the executables
@@ -37,25 +37,25 @@
 # These can be overridden by command line switches (see smbd(8))
 # or in smb.conf (see smb.conf(5))
 LOGFILEBASE = $(VARDIR)
-SMBLOGFILE = $(LOGFILEBASE)/log.smb
-NMBLOGFILE = $(LOGFILEBASE)/log.nmb
+SMBLOGFILE = $(LOGFILEBASE)/log/samba/log.smb
+NMBLOGFILE = $(LOGFILEBASE)/log/samba/log.nmb
 CONFIGFILE = $(LIBDIR)/smb.conf
 LMHOSTSFILE = $(LIBDIR)/lmhosts
 DRIVERFILE = $(LIBDIR)/printers.def
 FORMSFILE = $(LIBDIR)/ntforms.def
 NTDRIVERSDIR = $(LIBDIR)
 PASSWD_PROGRAM = /bin/passwd
-SMB_PASSWD_FILE = $(BASEDIR)/private/smbpasswd
-SMB_PASSGRP_FILE = $(BASEDIR)/private/smbpassgrp
-SMB_GROUP_FILE = $(BASEDIR)/private/smbgroup
-SMB_ALIAS_FILE = $(BASEDIR)/private/smbalias
+SMB_PASSWD_FILE = $(LIBDIR)/smbpasswd
+SMB_PASSGRP_FILE = $(LIBDIR)/smbpassgrp
+SMB_GROUP_FILE = $(LIBDIR)/smbgroup
+SMB_ALIAS_FILE = $(LIBDIR)/smbalias
 SMB_PASSWD_PROGRAM = $(BINDIR)/smbpasswd
 
 # This is where SWAT images and help files go
-SWATDIR = $(BASEDIR)/swat
+SWATDIR = $(BASEDIR)/share/swat
 
 # the directory where lock files go
-LOCKDIR = $(VARDIR)/locks
+LOCKDIR = $(VARDIR)/lock/samba
 
 # The directory where code page definition files go
 CODEPAGEDIR = $(LIBDIR)/codepages
