diff -uNr samba-PVERSION/source/Makefile.in samba-kgc/source/Makefile.in
--- samba-PVERSION/source/Makefile.in	Wed Dec  2 14:35:18 1998
+++ samba-kgc/source/Makefile.in	Wed Dec  2 15:06:02 1998
@@ -27,7 +27,7 @@
 # the previous releases of Samba
 SBINDIR = @bindir@
 LIBDIR = @libdir@
-VARDIR = $(BASEDIR)/var
+VARDIR = /var
 MANDIR = @mandir@
 
 # The permissions to give the executables
@@ -36,23 +36,23 @@
 # set these to where to find various files
 # These can be overridden by command line switches (see smbd(8))
 # or in smb.conf (see smb.conf(5))
-SMBLOGFILE = $(VARDIR)/log.smb
-NMBLOGFILE = $(VARDIR)/log.nmb
+SMBLOGFILE = $(VARDIR)/log/samba/log.smb
+NMBLOGFILE = $(VARDIR)/log/samba/log.nmb
 CONFIGFILE = $(LIBDIR)/smb.conf
 LMHOSTSFILE = $(LIBDIR)/lmhosts
 DRIVERFILE = $(LIBDIR)/printers.def
 PASSWD_PROGRAM = /bin/passwd
-SMB_PASSWD_FILE = $(BASEDIR)/private/smbpasswd
-SMB_PASSGRP_FILE = $(BASEDIR)/private/smbpassgrp
-SMB_GROUP_FILE = $(BASEDIR)/private/smbgroup
-SMB_ALIAS_FILE = $(BASEDIR)/private/smbalias
+SMB_PASSWD_FILE = $(LIBDIR)/smbpasswd
+SMB_PASSGRP_FILE = $(LIBDIR)/smbpassgrp
+SMB_GROUP_FILE = $(LIBDIR)/smbgroup
+SMB_ALIAS_FILE = $(LIBDIR)/smbalias
 SMB_PASSWD_PROGRAM = $(BINDIR)/smbpasswd
 
 # This is where SWAT images and help files go
-SWATDIR = $(BASEDIR)/swat
+SWATDIR = $(BASEDIR)/share/swat
 
 # the directory where lock files go
-LOCKDIR = $(VARDIR)/locks
+LOCKDIR = $(VARDIR)/lock/samba
 
 # The directory where code page definition files go
 CODEPAGEDIR = $(LIBDIR)/codepages
