--- samba-1.9.18p4/source/Makefile.orig	Fri Mar 20 09:09:25 1998
+++ samba-1.9.18p4/source/Makefile	Fri Mar 27 19:38:50 1998
@@ -5,11 +5,11 @@
 ###########################################################################
 
 # The base directory for all samba files
-BASEDIR = /usr/local/samba
+BASEDIR = /usr
 
 # The base manpages directory to put the man pages in
 # Note: $(MANDIR)/man1, $(MANDIR)/man5 and $(MANDIR)/man8 must exist.
-MANDIR = /usr/local/man
+MANDIR = /usr/man
 
 # The directories to put things in. If you use multiple
 # architectures or share the samba binaries across NFS then
@@ -18,9 +18,9 @@
 #       normally only applies to nmbd and smbd
 #       SBINDIR implies a secure binary directory
 BINDIR = $(BASEDIR)/bin
-SBINDIR = $(BASEDIR)/bin
-LIBDIR = $(BASEDIR)/lib
-VARDIR = $(BASEDIR)/var
+SBINDIR = $(BASEDIR)/sbin
+LIBDIR = /etc
+VARDIR = /var/log/samba
 
 # The permissions to give the executables
 INSTALLPERMS = 0755
@@ -32,7 +32,7 @@
 
 # You will need to use a ANSI C compiler. This means under SunOS 4 you can't 
 # use cc, instead you will have to use gcc. 
-# CC = gcc
+CC = gcc
 
 # This may help with some versions of make
 SHELL = /bin/sh
@@ -51,11 +51,11 @@
 LMHOSTSFILE = $(LIBDIR)/lmhosts
 DRIVERFILE = $(LIBDIR)/printers.def
 SMB_PASSWD = $(BINDIR)/smbpasswd
-SMB_PASSWD_FILE = $(BASEDIR)/private/smbpasswd
+SMB_PASSWD_FILE = $(LIBDIR)/smbpasswd
 WEB_ROOT = $(BASEDIR)
 
 # the directory where lock files go
-LOCKDIR = $(VARDIR)/locks
+LOCKDIR = /var/lock/samba
 
 # The directory where code page definition files go
 CODEPAGEDIR = $(LIBDIR)/codepages
@@ -90,8 +90,8 @@
 
 # This is for PAM authentication. RedHat Linux uses PAM.
 # If you use PAM, then uncomment the following lines:
-# PAM_FLAGS = -DUSE_PAM
-# PAM_LIBS = -ldl -lpam
+PAM_FLAGS = -DUSE_PAM
+PAM_LIBS = -ldl -lpam
 
 # This is for AFS authentication.  If you use AFS then set AFS_BASE 
 # according to your system layout, and uncomment the other lines as well.
@@ -215,8 +215,8 @@
 # contributed by Andrew.Tridgell@anu.edu.au & John.Terpstra@aquasoft.com.au
 # AXPROC defines DEC Alpha Processor
 # FLAGSM = -DLINUX -DAXPROC -DFAST_SHARE_MODES
-# FLAGSM = -DLINUX -DFAST_SHARE_MODES
-# LIBSM = 
+FLAGSM = -DLINUX -DFAST_SHARE_MODES
+LIBSM = 
 
 # Use this for Linux with shadow passwords and quota - but not using PAM!
 # contributed by xeno@mix.hive.no
