--- examples/pdb/Makefile	Thu Sep  5 02:11:41 2002
+++ examples/pdb/Makefile	Thu Sep  5 02:11:59 2002
@@ -8,7 +8,7 @@
 SAMBA_INCL = ../../source/include
 UBIQX_SRC = ../../source/ubiqx
 SMBWR_SRC = ../../source/smbwrapper
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -Wall -g
+CFLAGS = -I/usr/include/heimdal -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -Wall -g
 PDB_OBJS = pdb_test.so
 
 # Default target
