--- source/include/ads.h	2003-06-10 08:51:03.000000000 +0200
+++ source/include/ads.h	2003-06-22 23:14:24.000000000 +0200
@@ -215,9 +215,11 @@
 #define ENCTYPE_ARCFOUR_HMAC ENCTYPE_ARCFOUR_HMAC_MD5
 #endif
 
+#if 0
 /* The older versions of heimdal that don't have this
    define don't seem to use it anyway.  I'm told they
    always use a subkey */
 #ifndef AP_OPTS_USE_SUBKEY
 #define AP_OPTS_USE_SUBKEY 0
 #endif
+#endif
