--- source/Makefile.in	2003-04-23 10:43:06.000000000 +0200
+++ source/Makefile.in	2003-04-23 10:45:39.000000000 +0200
@@ -673,6 +673,13 @@
 	@echo "Using FLAGS = $(FLAGS)"
 	@echo "      FLAGS32 = $(FLAGS32)"
 	@echo "      LIBS = $(LIBS)"
+	@echo "      TERMLIBS = $(TERMLIBS)"
+	@echo "      PRINTLIBS = $(PRINTLIBS)"
+	@echo "      AUTHLIBS = $(AUTHLIBS)"
+	@echo "      ACLLIBS = $(ACLLIBS)"
+	@echo "      PASSDBLIBS = $(PASSDBLIBS)"
+	@echo "      ADSLIBS = $(ADSLIBS)"
+	@echo "      KRB5LIBS = $(KRB5_LIBS)"
 	@echo "      LDSHFLAGS = $(LDSHFLAGS)"
 	@echo "      LDFLAGS = $(LDFLAGS)"
 
