--- source/libads/kerberos.c	23 Oct 2002 00:02:26 -0000	1.18
+++ source/libads/kerberos.c	1 Jul 2003 21:30:17 -0000
@@ -126,6 +126,7 @@
 		return KRB5_LIBOS_CANTREADPWD;
 	}
 	
+	initialize_krb5_error_table();
 	ret = kerberos_kinit_password(s, ads->auth.password, ads->auth.time_offset);
 
 	if (ret) {
--- source/libads/kerberos_verify.c	6 Jun 2003 14:53:22 -0000	1.10.2.1
+++ source/libads/kerberos_verify.c	1 Jul 2003 21:30:17 -0000
@@ -62,6 +62,7 @@
 	password.data = password_s;
 	password.length = strlen(password_s);
 
+	initialize_krb5_error_table();
 	ret = krb5_init_context(&context);
 	if (ret) {
 		DEBUG(1,("krb5_init_context failed (%s)\n", error_message(ret)));
--- source/libads/krb5_setpw.c	6 Jun 2003 14:53:22 -0000	1.16.2.1
+++ source/libads/krb5_setpw.c	1 Jul 2003 21:30:17 -0000
@@ -470,6 +470,7 @@
 	krb5_creds creds, *credsp;
 	krb5_ccache ccache;
 
+	initialize_krb5_error_table();
 	ret = krb5_init_context(&context);
 	if (ret) {
 		DEBUG(1,("Failed to init krb5 context (%s)\n", error_message(ret)));
@@ -584,6 +585,7 @@
     krb5_creds creds;
     char *chpw_princ = NULL, *password;
 
+    initialize_krb5_error_table();
     ret = krb5_init_context(&context);
     if (ret) {
 	DEBUG(1,("Failed to init krb5 context (%s)\n", error_message(ret)));
--- source/libsmb/clikrb5.c	1 Jul 2003 14:40:37 -0000	1.36.2.2
+++ source/libsmb/clikrb5.c	1 Jul 2003 21:30:20 -0000
@@ -320,7 +320,8 @@
 		ENCTYPE_DES_CBC_MD5, 
 		ENCTYPE_DES_CBC_CRC, 
 		ENCTYPE_NULL};
-	
+
+	initialize_krb5_error_table();
 	retval = krb5_init_context(&context);
 	if (retval) {
 		DEBUG(1,("krb5_init_context failed (%s)\n", 
--- source/utils/net_lookup.c	1 Jul 2003 14:40:47 -0000	1.8.2.1
+++ source/utils/net_lookup.c	1 Jul 2003 21:30:24 -0000
@@ -177,6 +177,7 @@
 	krb5_data realm;
 	char **realms;
 
+	initialize_krb5_error_table();
 	rc = krb5_init_context(&ctx);
 	if (rc) {
 		DEBUG(1,("krb5_init_context failed (%s)\n", 
