--- examples/VFS/samba-vscan-0.3.2b/fprot/Makefile	2003-01-14 00:42:15.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/fprot/Makefile	2003-04-09 20:21:37.000000000 +0200
@@ -14,7 +14,7 @@
 SMBWR_SRC = ../../../../source/smbwrapper
 SMBVS_INCL = ../include
 SMBVS_GLB = ../global
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC 
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC -I/usr/include/heimdal 
 VFS_OBJS = vscan-fprotd.so
 SOURCES = $(SMBVS_GLB)/vscan-functions.c $(SMBVS_GLB)/vscan-message.c $(SMBVS_GLB)/vscan-quarantine.c $(SMBVS_GLB)/vscan-fileaccesslog.c vscan-fprotd.c vscan-fprotd_core.c vscan-fprotd.h vscan-fprotd_core.h
 OBJS = vscan-functions.lo vscan-message.lo vscan-quarantine.lo vscan-fileaccesslog.lo vscan-fprotd.lo vscan-fprotd_core.lo
--- examples/VFS/samba-vscan-0.3.2b/fprot/vscan-fprotd.c	2003-02-21 21:37:44.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/fprot/vscan-fprotd.c	2003-04-09 20:25:25.000000000 +0200
@@ -432,14 +432,14 @@
 				rc = vscan_do_infected_file_action(&default_vfs_ops, conn, filepath, quarantine_dir, quarantine_prefix, infected_file_action);
 
 				/* add/update file. mark file as infected! */
-                                lrufiles_add(filepath, stat_buf.st_mtime, TRUE);
+                                lrufiles_add(filepath, stat_buf.st_mtime, True);
 
 				/* virus found, deny acces */
 				errno = EACCES;
 				return -1;
 			} else if ( retval == 0 ) {
 				/* file is clean, add to lrufiles */
-				lrufiles_add(filepath, stat_buf.st_mtime, FALSE);
+				lrufiles_add(filepath, stat_buf.st_mtime, False);
 			}
 		}
 
--- examples/VFS/samba-vscan-0.3.2b/fprot/vscan-fprotd_core.c	2003-01-25 18:40:57.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/fprot/vscan-fprotd_core.c	2003-04-09 20:23:31.000000000 +0200
@@ -110,7 +110,7 @@
 	pstring fprotdCommand;	/* the command line to be send to daemon */
 	char *str;
 	FILE *fpin, *fpout;
-	bool received_data = FALSE; /* indicates, if any response from deamon was received */
+	bool received_data = False; /* indicates, if any response from deamon was received */
 
 	/* open stream sockets */
         fpin = fdopen(sockfd, "r");
@@ -159,7 +159,7 @@
 
 	while ( (fgets(recvline, MAXLINE, fpin)) != NULL ) {
 
-		received_data = TRUE;
+		received_data = True;
 
 		/* ignore the HTTP response header, remove any leading 
 		   white spaces */
--- examples/VFS/samba-vscan-0.3.2b/icap/Makefile	2003-01-30 00:53:02.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/icap/Makefile	2003-04-09 20:21:37.000000000 +0200
@@ -15,7 +15,7 @@
 SMBWR_SRC = ../../../../source/smbwrapper
 SMBVS_INCL = ../include
 SMBVS_GLB = ../global
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC -I/usr/include/heimdal
 VFS_OBJS = vscan-icap.so
 SOURCES = $(SMBVS_GLB)/vscan-functions.c $(SMBVS_GLB)/vscan-message.c $(SMBVS_GLB)/vscan-quarantine.c $(SMBVS_GLB)/vscan-fileaccesslog.c vscan-icap.c vscan-icap_core.c vscan-icap.h vscan-icap_core.h
 OBJS = vscan-functions.lo vscan-message.lo vscan-quarantine.lo vscan-fileaccesslog.lo vscan-icap.lo vscan-icap_core.lo
--- examples/VFS/samba-vscan-0.3.2b/icap/vscan-icap.c	2003-02-21 21:37:50.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/icap/vscan-icap.c	2003-04-09 20:32:20.000000000 +0200
@@ -413,14 +413,14 @@
                                 rc = vscan_do_infected_file_action(&default_vfs_ops, conn, filepath, quarantine_dir, quarantine_prefix, infected_file_action);
 
                                /* add/update file. mark file as infected! */
-                                lrufiles_add(filepath, stat_buf.st_mtime, TRUE);
+                                lrufiles_add(filepath, stat_buf.st_mtime, True);
 
                                 /* virus found, deny acces */
                                 errno = EACCES;
                                 return -1;
 			} else if ( retval == 0 ) {
                                 /* file is clean, add to lrufiles */
-                                lrufiles_add(filepath, stat_buf.st_mtime, FALSE);
+                                lrufiles_add(filepath, stat_buf.st_mtime, False);
                         }
                 }
 
--- examples/VFS/samba-vscan-0.3.2b/icap/vscan-icap_core.c	2003-01-15 00:19:18.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/icap/vscan-icap_core.c	2003-04-09 20:30:56.000000000 +0200
@@ -114,8 +114,8 @@
         char buf[BUFLEN];
         char recvline[MAXLINE + 1];
 	char *str;
-       	bool first_line = FALSE; /* first line we've received? */
-        bool infected = FALSE;	/* an infected found? */
+       	bool first_line = False; /* first line we've received? */
+        bool infected = False;	/* an infected found? */
 
 
 	/* get file length */
@@ -213,7 +213,7 @@
 	/* set line buffering */
         setvbuf(fpin, (char *)NULL, _IOLBF, 0);
 
-        first_line = TRUE;
+        first_line = True;
         while ( (fgets(recvline, MAXLINE, fpin)) != NULL ) {
 		str = recvline;
 		if ( first_line ) {
@@ -226,7 +226,7 @@
 						return(0);
                                         }
                                         else if ( strncmp("403", str, 3) == 0 ) {
-                                                infected = TRUE;
+                                                infected = True;
                                         } else {
 						if ( verbose_file_logging )
 							vscan_syslog("ERROR: file %s not found, not readable or an error occured", scan_file);
@@ -241,7 +241,7 @@
 				return(-1);
                         }
 
-                        first_line = FALSE;
+                        first_line = False;
                 }
 		if ( infected ) {
 			if ( strncmp("X-Infection-Found", str, 17) == 0 ) {
--- examples/VFS/samba-vscan-0.3.2b/include/vscan-global.h	2002-11-25 16:48:10.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/include/vscan-global.h	2003-04-09 20:21:37.000000000 +0200
@@ -93,7 +93,7 @@
 */
 
 #ifndef SAMBA_VERSION_MAJOR
-# define SAMBA_VERSION_MAJOR 2 
+# define SAMBA_VERSION_MAJOR 3 
 #endif
 
 #ifndef SAMBA_VERSION_MINOR
--- examples/VFS/samba-vscan-0.3.2b/kaspersky/Makefile	2003-02-20 15:41:32.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/kaspersky/Makefile	2003-04-09 20:21:37.000000000 +0200
@@ -23,9 +23,9 @@
 VFS_OBJS = vscan-kavp.so
 
 ifdef USE_DEBUG
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC 
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC -I/usr/include/heimdal
 else
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC -I/usr/include/heimdal 
 endif
 
 ifndef USE_KAVPSHAREDLIB
--- examples/VFS/samba-vscan-0.3.2b/mks/Makefile	2003-01-19 18:09:53.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/mks/Makefile	2003-04-09 20:21:37.000000000 +0200
@@ -16,10 +16,10 @@
 SMBWR_SRC = ../../../../source/smbwrapper
 SMBVS_INCL = ../include
 SMBVS_GLB = ../global
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC 
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC -I/usr/include/heimdal 
 
 ifdef USE_INCLMKSDLIB
-CFLAGS=-I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -D_USE_INCL_MKSD_LIB=1 -fPIC
+CFLAGS=-I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -D_USE_INCL_MKSD_LIB=1 -fPIC -I/usr/include/heimdal
 endif
 
 VFS_OBJS = vscan-mksd.so
--- examples/VFS/samba-vscan-0.3.2b/mks/vscan-mksd.c	2003-02-21 21:37:50.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/mks/vscan-mksd.c	2003-04-09 20:38:16.000000000 +0200
@@ -393,14 +393,14 @@
 				rc = vscan_do_infected_file_action(&default_vfs_ops, conn, filepath, quarantine_dir, quarantine_prefix, infected_file_action);
 
                                 /* add/update file. mark file as infected! */
-                                lrufiles_add(filepath, stat_buf.st_mtime, TRUE);
+                                lrufiles_add(filepath, stat_buf.st_mtime, True);
 
 				/* virus found, deny acces */
 				errno = EACCES; 
 				return -1;
                         } else if ( retval == 0 ) {
                                 /* file is clean, add to lrufiles */
-                                lrufiles_add(filepath, stat_buf.st_mtime, FALSE);
+                                lrufiles_add(filepath, stat_buf.st_mtime, False);
 			}
 		}
 
--- examples/VFS/samba-vscan-0.3.2b/openantivirus/Makefile	2003-01-30 00:53:08.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/openantivirus/Makefile	2003-04-09 20:21:37.000000000 +0200
@@ -15,7 +15,7 @@
 SMBWR_SRC = ../../../../source/smbwrapper
 SMBVS_INCL = ../include
 SMBVS_GLB = ../global
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC 
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC -I/usr/include/heimdal
 VFS_OBJS = vscan-oav.so
 SOURCES = $(SMBVS_GLB)/vscan-functions.c $(SMBVS_GLB)/vscan-message.c $(SMBVS_GLB)/vscan-quarantine.c $(SMBVS_GLB)/vscan-fileaccesslog.c vscan-oav.c vscan-oav_core.c vscan-oav.h vscan-oav_core.h
 OBJS = vscan-functions.lo vscan-message.lo vscan-quarantine.lo vscan-fileaccesslog.lo vscan-oav.lo vscan-oav_core.lo
--- examples/VFS/samba-vscan-0.3.2b/openantivirus/vscan-oav.c	2003-02-21 21:37:51.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/openantivirus/vscan-oav.c	2003-04-09 20:40:53.000000000 +0200
@@ -417,14 +417,14 @@
 				rc = vscan_do_infected_file_action(&default_vfs_ops, conn, filepath, quarantine_dir, quarantine_prefix, infected_file_action);
 
                                 /* add/update file. mark file as infected! */
-                                lrufiles_add(filepath, stat_buf.st_mtime, TRUE);
+                                lrufiles_add(filepath, stat_buf.st_mtime, True);
 
 				/* virus found, deny acces */
 				errno = EACCES; 
 				return -1;
                         } else if ( retval == 0 ) {
                                 /* file is clean, add to lrufiles */
-                                lrufiles_add(filepath, stat_buf.st_mtime, FALSE);
+                                lrufiles_add(filepath, stat_buf.st_mtime, False);
                         }
 		}
 
--- examples/VFS/samba-vscan-0.3.2b/sophos/Makefile	2003-01-30 00:53:08.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/sophos/Makefile	2003-04-09 20:21:37.000000000 +0200
@@ -15,7 +15,7 @@
 SMBWR_SRC = ../../../../source/smbwrapper
 SMBVS_INCL = ../include
 SMBVS_GLB = ../global
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC 
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC -I/usr/include/heimdal
 VFS_OBJS = vscan-sophos.so
 SOURCES = $(SMBVS_GLB)/vscan-functions.c $(SMBVS_GLB)/vscan-message.c $(SMBVS_GLB)/vscan-quarantine.c $(SMBVS_GLB)/vscan-fileaccesslog.c vscan-sophos.c vscan-sophos_core.c vscan-sophos.h vscan-sophos_core.h
 OBJS = vscan-functions.lo vscan-message.lo vscan-quarantine.lo vscan-fileaccesslog.lo vscan-sophos.lo vscan-sophos_core.lo
--- examples/VFS/samba-vscan-0.3.2b/sophos/vscan-sophos.c	2003-02-21 21:37:51.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/sophos/vscan-sophos.c	2003-04-09 20:43:11.000000000 +0200
@@ -399,14 +399,14 @@
 				rc = vscan_do_infected_file_action(&default_vfs_ops, conn, filepath, quarantine_dir, quarantine_prefix, infected_file_action);
 
                                 /* add/update file. mark file as infected! */
-                                lrufiles_add(filepath, stat_buf.st_mtime, TRUE);
+                                lrufiles_add(filepath, stat_buf.st_mtime, True);
 
 				/* deny access */
 				errno = EACCES; 
 				return -1;
                         } else if ( retval == 0 ) {
                                 /* file is clean, add to lrufiles */
-                                lrufiles_add(filepath, stat_buf.st_mtime, FALSE);
+                                lrufiles_add(filepath, stat_buf.st_mtime, False);
                         }
 		}
 
--- examples/VFS/samba-vscan-0.3.2b/trend/Makefile	2003-01-30 01:03:38.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/trend/Makefile	2003-04-09 20:21:37.000000000 +0200
@@ -15,7 +15,7 @@
 SMBWR_SRC = ../../../../source/smbwrapper
 SMBVS_INCL = ../include
 SMBVS_GLB = ../global
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC 
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -fPIC -I/usr/include/heimdal
 VFS_OBJS = vscan-trend.so
 SOURCES = $(SMBVS_GLB)/vscan-functions.c $(SMBVS_GLB)/vscan-message.c $(SMBVS_GLB)/vscan-quarantine.c $(SMBVS_GLB)/vscan-fileaccesslog.c vscan-trend.c vscan-trend_core.c vscan-trend.h vscan-trend_core.h
 OBJS = vscan-functions.lo vscan-message.lo vscan-quarantine.lo vscan-fileaccesslog.lo vscan-trend.lo vscan-trend_core.lo
--- examples/VFS/samba-vscan-0.3.2b/trend/vscan-trend.c	2003-02-21 21:37:52.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.2b/trend/vscan-trend.c	2003-04-09 20:46:07.000000000 +0200
@@ -409,14 +409,14 @@
 				rc = vscan_do_infected_file_action(&default_vfs_ops, conn, filepath, quarantine_dir, quarantine_prefix, infected_file_action);
 
                                 /* add/update file. mark file as infected! */
-                                lrufiles_add(filepath, stat_buf.st_mtime, TRUE);
+                                lrufiles_add(filepath, stat_buf.st_mtime, True);
 
 				/* deny access */
 				errno = EACCES; 
 				return -1;
                         } else if ( retval == 0 ) {
                                 /* file is clean, add to lrufiles */
-                                lrufiles_add(filepath, stat_buf.st_mtime, FALSE);
+                                lrufiles_add(filepath, stat_buf.st_mtime, False);
                         }
 		}
 
