--- examples/VFS/samba-vscan-0.3.1/fprot/Makefile	2002-11-26 15:20:17.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.1/fprot/Makefile	2002-12-19 13:26:19.000000000 +0100
@@ -14,7 +14,7 @@
 SMBWR_SRC = ../../../../source/smbwrapper
 SMBVS_INCL = ../include
 SMBVS_GLB = ../global
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/heimdal 
 VFS_OBJS = vscan-fprotd.so
 SOURCES = $(SMBVS_GLB)/vscan-functions.c $(SMBVS_GLB)/vscan-message.c $(SMBVS_GLB)/vscan-quarantine.c vscan-fprotd.c vscan-fprotd_core.c vscan-fprotd.h vscan-fprotd_core.h
 OBJS = vscan-functions.lo vscan-message.lo vscan-quarantine.lo vscan-fprotd.lo vscan-fprotd_core.lo
--- examples/VFS/samba-vscan-0.3.1/include/vscan-global.h	2002-11-25 16:48:10.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.1/include/vscan-global.h	2002-12-19 13:26:34.000000000 +0100
@@ -93,7 +93,7 @@
 */
 
 #ifndef SAMBA_VERSION_MAJOR
-# define SAMBA_VERSION_MAJOR 2 
+# define SAMBA_VERSION_MAJOR 3 
 #endif
 
 #ifndef SAMBA_VERSION_MINOR
--- examples/VFS/samba-vscan-0.3.1/kaspersky/Makefile	2002-11-28 17:40:35.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.1/kaspersky/Makefile	2002-12-19 13:27:23.000000000 +0100
@@ -23,9 +23,9 @@
 VFS_OBJS = vscan-kavp.so
 
 ifdef USE_DEBUG
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/heimdal
 else
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/heimdal 
 endif
 
 ifndef USE_KAVPSHAREDLIB
--- examples/VFS/samba-vscan-0.3.1/mks/Makefile	2002-11-26 16:29:55.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.1/mks/Makefile	2002-12-19 13:27:53.000000000 +0100
@@ -13,7 +13,7 @@
 SMBWR_SRC = ../../../../source/smbwrapper
 SMBVS_INCL = ../include
 SMBVS_GLB = ../global
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/heimdal 
 VFS_OBJS = vscan-mksd.so
 SOURCES = $(SMBVS_GLB)/vscan-functions.c $(SMBVS_GLB)/vscan-message.c $(SMBVS_GLB)/vscan-quarantine.c  vscan-mksd.c vscan-mksd_core.c vscan-mksd.h vscan-mksd_core.h mks.h mks_c.c
 OBJS = vscan-functions.lo vscan-message.lo vscan-quarantine.lo vscan-mksd.lo vscan-mksd_core.lo mks_c.lo
--- examples/VFS/samba-vscan-0.3.1/openantivirus/Makefile	2002-11-27 19:24:03.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.1/openantivirus/Makefile	2002-12-19 13:28:10.000000000 +0100
@@ -15,7 +15,7 @@
 SMBWR_SRC = ../../../../source/smbwrapper
 SMBVS_INCL = ../include
 SMBVS_GLB = ../global
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/heimdal 
 VFS_OBJS = vscan-oav.so
 SOURCES = $(SMBVS_GLB)/vscan-functions.c $(SMBVS_GLB)/vscan-message.c $(SMBVS_GLB)/vscan-quarantine.c  vscan-oav.c vscan-oav_core.c vscan-oav.h vscan-oav_core.h
 OBJS = vscan-functions.lo vscan-message.lo vscan-quarantine.lo vscan-oav.lo vscan-oav_core.lo
--- examples/VFS/samba-vscan-0.3.1/sophos/Makefile	2002-11-27 19:24:03.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.1/sophos/Makefile	2002-12-19 13:29:20.000000000 +0100
@@ -15,7 +15,7 @@
 SMBWR_SRC = ../../../../source/smbwrapper
 SMBVS_INCL = ../include
 SMBVS_GLB = ../global
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/heimdal 
 VFS_OBJS = vscan-sophos.so
 SOURCES = $(SMBVS_GLB)/vscan-functions.c $(SMBVS_GLB)/vscan-message.c $(SMBVS_GLB)/vscan-quarantine.c vscan-sophos.c vscan-sophos_core.c vscan-sophos.h vscan-sophos_core.h
 OBJS = vscan-functions.lo vscan-message.lo vscan-quarantine.lo vscan-sophos.lo vscan-sophos_core.lo
--- examples/VFS/samba-vscan-0.3.1/trend/Makefile	2002-11-27 19:24:03.000000000 +0100
+++ examples/VFS/samba-vscan-0.3.1/trend/Makefile	2002-12-19 13:29:31.000000000 +0100
@@ -15,7 +15,7 @@
 SMBWR_SRC = ../../../../source/smbwrapper
 SMBVS_INCL = ../include
 SMBVS_GLB = ../global
-CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64  
+CFLAGS = -I$(SAMBA_SRC) -I$(SAMBA_INCL) -I$(UBIQX_SRC) -I$(SMBWR_SRC) -I$(SMBVS_INCL) -Wall -g -D_LARGEFILE64_SOURCE -D_FILE_OFFSET_BITS=64 -I/usr/include/heimdal 
 VFS_OBJS = vscan-trend.so
 SOURCES = $(SMBVS_GLB)/vscan-functions.c $(SMBVS_GLB)/vscan-message.c $(SMBVS_GLB)/vscan-quarantine.c vscan-trend.c vscan-trend_core.c vscan-trend.h vscan-trend_core.h
 OBJS = vscan-functions.lo vscan-message.lo vscan-quarantine.lo vscan-trend.lo vscan-trend_core.lo
