--- samba-2.2.0/source/script/installbin.sh	Fri Jan 12 21:28:02 2001
+++ samba-2.2.0/source/script/installbin.sh.207	Thu Apr 19 09:40:57 2001
@@ -33,9 +33,9 @@
  chmod $INSTALLPERMS $BINDIR/$p2
 
  # this is a special case, mount needs this in a specific location
- if [ $p2 = smbmount ]; then
-   ln -sf $BINDIR/$p2 /sbin/mount.smbfs
- fi
+ #if [ $p2 = smbmount ]; then
+ #  ln -sf $BINDIR/$p2 /sbin/mount.smbfs
+ #fi
 done
 
 
