--- samba-3.0.2a/examples/LDAP/smbldap-tools/smbldap_conf.pm.orig	2003-12-10 22:59:17.000000000 +0100
+++ samba-3.0.2a/examples/LDAP/smbldap-tools/smbldap_conf.pm	2004-03-01 22:59:01.535482375 +0100
@@ -101,22 +101,22 @@
 
 # LDAP Suffix
 # Ex: $suffix = "dc=IDEALX,dc=ORG";
-$suffix = "dc=IDEALX,dc=COM";
+$suffix = "dc=example,dc=com";
 
 
 # Where are stored Users
 # Ex: $usersdn = "ou=Users,$suffix"; for ou=Users,dc=IDEALX,dc=ORG
-$usersou = q(_USERS_);
+$usersou = q(People);
 $usersdn = "ou=$usersou,$suffix";
 
 # Where are stored Computers
 # Ex: $computersdn = "ou=Computers,$suffix"; for ou=Computers,dc=IDEALX,dc=ORG
-$computersou = q(_COMPUTERS_);
+$computersou = q(Hosts);
 $computersdn = "ou=$computersou,$suffix";
 
 # Where are stored Groups
 # Ex $groupsdn = "ou=Groups,$suffix"; for ou=Groups,dc=IDEALX,dc=ORG
-$groupsou = q(_GROUPS_);
+$groupsou = q(Group);
 $groupsdn = "ou=$groupsou,$suffix";
 
 # Default scope Used
@@ -130,7 +130,7 @@
 ############################
 # Bind DN used 
 # Ex: $binddn = "cn=Manager,$suffix"; for cn=Manager,dc=IDEALX,dc=org
-$binddn = "cn=Manager,$suffix";
+$binddn = "cn=root,$suffix";
 
 # Bind DN passwd used
 # Ex: $bindpasswd = 'secret'; for 'secret'
@@ -153,11 +153,11 @@
 # Login defs
 # Default Login Shell
 # Ex: $_userLoginShell = q(/bin/bash);
-$_userLoginShell = q(_LOGINSHELL_);
+$_userLoginShell = q(/bin/bash);
 
 # Home directory prefix (without username)
 # Ex: $_userHomePrefix = q(/home/);
-$_userHomePrefix = q(_HOMEPREFIX_);
+$_userHomePrefix = q(/home);
 
 # Gecos
 $_userGecos = q(System User);
@@ -187,19 +187,19 @@
 # Ex: q(\\\\My-PDC-netbios-name\\homes) for \\My-PDC-netbios-name\homes
 # Just comment this if you want to use the smb.conf 'logon home' directive
 # and/or desabling roaming profiles
-$_userSmbHome = q(\\\\_PDCNAME_\\homes);
+#$_userSmbHome = q(\\\\_PDCNAME_\\homes);
 
 # The UNC path to profiles locations without the username last extension
 # (will be dynamically prepended)
 # Ex: q(\\\\My-PDC-netbios-name\\profiles\\) for \\My-PDC-netbios-name\profiles
 # Just comment this if you want to use the smb.conf 'logon path' directive
 # and/or desabling roaming profiles
-$_userProfile = q(\\\\_PDCNAME_\\profiles\\);
+#$_userProfile = q(\\\\_PDCNAME_\\profiles\\);
 
 # The default Home Drive Letter mapping
 # (will be automatically mapped at logon time if home directory exist)
 # Ex: q(U:) for U:
-$_userHomeDrive = q(_HOMEDRIVE_);
+#$_userHomeDrive = q(_HOMEDRIVE_);
 
 # The default user netlogon script name
 # if not used, will be automatically username.cmd
@@ -216,7 +216,7 @@
 # prefer mkntpwd... most of the time, it's a wise choice :-) 
 $with_smbpasswd = 0;
 $smbpasswd = "/usr/bin/smbpasswd";
-$mk_ntpasswd = "/usr/local/sbin/mkntpwd";
+$mk_ntpasswd = "/usr/sbin/mkntpwd";
 
 # those next externals commands are kept fot the migration scripts and
 # for the populate script: this will be updated as soon as possible
