*** samba-1.9.16p11/source/Makefile.RPM	
--- samba-1.9.16p11/source/Makefile
***************
*** 179,186 ****
  # Use this for Linux without shadow passwords
  # contributed by Andrew.Tridgell@anu.edu.au
  # AXPROC defines DEC Alpha Processor
! # FLAGSM = -DLINUX -DAXPROC
! FLAGSM = -DLINUX
  LIBSM = 
  
  # Use this for Linux with shadow passwords and quota
--- 179,186 ----
  # Use this for Linux without shadow passwords
  # contributed by Andrew.Tridgell@anu.edu.au
  # AXPROC defines DEC Alpha Processor
! FLAGSM = -DLINUX -DAXPROC
! # FLAGSM = -DLINUX
  LIBSM = 
  
  # Use this for Linux with shadow passwords and quota
