*** samba-1.9.17/source/Makefile.orig	Sat Jul 12 12:51:39 1997
--- samba-1.9.17/source/Makefile	Sat Jul 12 12:56:04 1997
***************
*** 5,15 ****
  ###########################################################################
  
  # The base directory for all samba files
! BASEDIR = /usr/local/samba
  
  # The base manpages directory to put the man pages in
  # Note: $(MANDIR)/man1, $(MANDIR)/man5 and $(MANDIR)/man8 must exist.
! MANDIR = /usr/local/man
  
  # The directories to put things in. If you use multiple
  # architectures or share the samba binaries across NFS then
--- 5,15 ----
  ###########################################################################
  
  # The base directory for all samba files
! BASEDIR = /usr
  
  # The base manpages directory to put the man pages in
  # Note: $(MANDIR)/man1, $(MANDIR)/man5 and $(MANDIR)/man8 must exist.
! MANDIR = $(BASEDIR)/man
  
  # The directories to put things in. If you use multiple
  # architectures or share the samba binaries across NFS then
***************
*** 18,26 ****
  #       normally only applies to nmbd and smbd
  #       SBINDIR implies a secure binary directory
  BINDIR = $(BASEDIR)/bin
! SBINDIR = $(BASEDIR)/bin
! LIBDIR = $(BASEDIR)/lib
! VARDIR = $(BASEDIR)/var
  
  # The permissions to give the executables
  INSTALLPERMS = 0755
--- 18,26 ----
  #       normally only applies to nmbd and smbd
  #       SBINDIR implies a secure binary directory
  BINDIR = $(BASEDIR)/bin
! SBINDIR = $(BASEDIR)/sbin
! LIBDIR = /etc
! VARDIR = /var
  
  # The permissions to give the executables
  INSTALLPERMS = 0755
***************
*** 35,41 ****
  
  # You will need to use a ANSI C compiler. This means under SunOS 4 you can't 
  # use cc, instead you will have to use gcc. 
! # CC = gcc
  
  # This may help with some versions of make
  SHELL = /bin/sh
--- 35,41 ----
  
  # You will need to use a ANSI C compiler. This means under SunOS 4 you can't 
  # use cc, instead you will have to use gcc. 
! CC = gcc
  
  # This may help with some versions of make
  SHELL = /bin/sh
***************
*** 48,60 ****
  # set these to where to find various files
  # These can be overridden by command line switches (see smbd(8))
  # or in smb.conf (see smb.conf(5))
! SMBLOGFILE = $(VARDIR)/log.smb
! NMBLOGFILE = $(VARDIR)/log.nmb
  CONFIGFILE = $(LIBDIR)/smb.conf
  LMHOSTSFILE = $(LIBDIR)/lmhosts
  
  # the directory where lock files go
! LOCKDIR = $(VARDIR)/locks
  
  # set this to the default group you want your machine to appear in
  # for browsing. This can also be set in nmbd (see nmbd(8))
--- 48,60 ----
  # set these to where to find various files
  # These can be overridden by command line switches (see smbd(8))
  # or in smb.conf (see smb.conf(5))
! SMBLOGFILE = $(VARDIR)/log/samba/log.smb
! NMBLOGFILE = $(VARDIR)/log/samba/log.nmb
  CONFIGFILE = $(LIBDIR)/smb.conf
  LMHOSTSFILE = $(LIBDIR)/lmhosts
  
  # the directory where lock files go
! LOCKDIR = $(VARDIR)/lock/samba
  
  # set this to the default group you want your machine to appear in
  # for browsing. This can also be set in nmbd (see nmbd(8))
***************
*** 77,84 ****
  
  # This is for PAM authentication. RedHat Linux uses PAM.
  # If you use PAM, then uncomment the following lines:
! # PAM_FLAGS = -DUSE_PAM
! # PAM_LIBS = -ldl -lpam
  
  # This is for AFS authentication.  If you use AFS then set AFS_BASE 
  # according to your system layout, and uncomment the other lines as well.
--- 77,84 ----
  
  # This is for PAM authentication. RedHat Linux uses PAM.
  # If you use PAM, then uncomment the following lines:
! PAM_FLAGS = -DUSE_PAM
! PAM_LIBS = -ldl -lpam
  
  # This is for AFS authentication.  If you use AFS then set AFS_BASE 
  # according to your system layout, and uncomment the other lines as well.
***************
*** 187,194 ****
  # contributed by Andrew.Tridgell@anu.edu.au
  # AXPROC defines DEC Alpha Processor
  # FLAGSM = -DLINUX -DAXPROC -DFAST_SHARE_MODES
! # FLAGSM = -DLINUX -DFAST_SHARE_MODES
! # LIBSM = 
  
  # Use this for Linux with shadow passwords and quota
  # contributed by xeno@mix.hive.no
--- 187,194 ----
  # contributed by Andrew.Tridgell@anu.edu.au
  # AXPROC defines DEC Alpha Processor
  # FLAGSM = -DLINUX -DAXPROC -DFAST_SHARE_MODES
! FLAGSM = -DLINUX -DFAST_SHARE_MODES
! LIBSM = 
  
  # Use this for Linux with shadow passwords and quota
  # contributed by xeno@mix.hive.no
