Index: Makefile.in
===================================================================
RCS file: /data/cvs/samba/source/Makefile.in,v
retrieving revision 1.465
diff -u -r1.465 Makefile.in
--- Makefile.in	2002/04/04 22:58:56	1.465
+++ Makefile.in	2002/04/05 05:48:51
@@ -785,6 +785,29 @@
 	-$(INSTALLCMD) -d ${prefix}/include
 	-$(INSTALLCMD) include/libsmbclient.h ${prefix}/include
 
+# Python extensions
+
+PYTHON_OBJS = $(LIB_OBJ) $(LIBSMB_OBJ) $(RPC_PARSE_OBJ) $(UBIQX_OBJ) \
+	$(PARAM_OBJ) $(LIBMSRPC_OBJ) $(PASSDB_OBJ) $(GROUPDB_OBJ)
+
+PY_SPOOLSS_PROTO_OBJ = python/py_spoolss.o \
+		python/py_spoolss_printers.o python/py_spoolss_printers_conv.o\
+		python/py_spoolss_forms.o python/py_spoolss_forms_conv.o \
+		python/py_spoolss_ports.o python/py_spoolss_ports_conv.o \
+		python/py_spoolss_drivers.o python/py_spoolss_drivers_conv.o \
+
+python_proto: python_spoolss_proto
+
+python_spoolss_proto:
+	@cd $(srcdir) && $(SHELL) script/mkproto.sh $(AWK) \
+		-h _PY_SPOOLSS_PROTO_H python/py_spoolss_proto.h \
+		$(PY_SPOOLSS_PROTO_OBJ)
+
+python_ext: $(PYTHON_OBJS)
+	@echo python python/setup.py build
+	@PYTHON_OBJS="$(PYTHON_OBJS)" PYTHON_CFLAGS="$(CFLAGS) $(CPPFLAGS)" \
+		python python/setup.py build
+
 # revert to the previously installed version
 revert:
 	@$(SHELL) $(srcdir)/script/revert.sh $(SBINDIR) $(SPROGS) 
Index: configure.in
===================================================================
RCS file: /data/cvs/samba/source/configure.in,v
retrieving revision 1.298
diff -u -r1.298 configure.in
--- configure.in	2002/04/04 05:47:41	1.298
+++ configure.in	2002/04/05 05:48:52
@@ -2695,7 +2695,7 @@
 builddir=`pwd`
 AC_SUBST(builddir)
 
-AC_OUTPUT(include/stamp-h Makefile)
+AC_OUTPUT(include/stamp-h Makefile python/setup.py)
 
 #################################################
 # Print very concise instructions on building/use
