--- utils/torture.c.00	Thu Jan 24 17:54:13 2002
+++ utils/torture.c	Thu Jan 24 16:58:52 2002
@@ -4801,8 +4801,13 @@
 	dbf = stdout;
 
 #ifdef HAVE_SETBUFFER
-	setbuffer(stdout, NULL, 0);
+        setbuffer(stdout, NULL, 0);
+#else
+#if defined(_SCO_DS) /* SCO OpenServer */
+	setvbuf(stdout, NULL, _IONBF, 0);
 #endif
+#endif
+
 	charset_initialise();
 		
 	codepage_initialise(lp_client_code_page());

--- utils/locktest.c.00	Fri Sep 29 13:18:14 2000
+++ utils/locktest.c	Fri Apr 13 17:54:11 2001
@@ -384,8 +384,12 @@
 			recorded[n].conn = random() % NCONNECTIONS;
 			recorded[n].f = random() % NFILES;
 			recorded[n].start = LOCKBASE + ((unsigned)random() % (LOCKRANGE-1));
+#if defined(_SCO_DS) /* OpenServer */
+			recorded[n].len = 1;
+#else
 			recorded[n].len = 1 + 
 				random() % (LOCKRANGE-(recorded[n].start-LOCKBASE));
+#endif
 			recorded[n].start *= RANGE_MULTIPLE;
 			recorded[n].len *= RANGE_MULTIPLE;
 			recorded[n].r1 = random() % 100;
